giving-example/src/main/java/com/adyen/giving/web/CheckoutController.java

```diff
@@ -0,0 +1,53 @@
+package com.adyen.giving.web;
+
+import com.adyen.giving.ApplicationProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestParam;
+
+@Controller
+public class CheckoutController {
+
+    @Autowired
+    private ApplicationProperty applicationProperty;
+
+    @Autowired
+    public CheckoutController(ApplicationProperty applicationProperty) {
+        this.applicationProperty = applicationProperty;
+
+        if(this.applicationProperty.getClientKey() == null) {
+            Logger log = LoggerFactory.getLogger(CheckoutController.class);
+            log.warn("ADYEN_CLIENT_KEY is undefined ");
+        }
+    }
+
+    @GetMapping("/")
+    public String index() {
+        return "index";
+    }
+
+    @GetMapping("/preview")
+    public String preview(@RequestParam String type, Model model) {
+        model.addAttribute("type", type);
```

Comment:
We pass `type` to the views but it is not necessary (we have one payment option so we can hardcode the div id in the view - as it is done for the donation DIV). I suggest to remove type everywhere (controller, view, jS)

Reply:
You have sharp eyes btw, I appreciate that